tests/typeclass-tests.dx

```diff
@@ -41,3 +41,81 @@ instance InterfaceTest4 Float
   foo = 1
   bar = \_. 1
 
+
+'# Synthesis errors
+
+'## Duplicate instances
+
+interface InterfaceSynthesisTest a
+  runTest : a
+
+instance InterfaceSynthesisTest Float
+  runTest = 1.0
+
+instance InterfaceSynthesisTest Float
+  runTest = 2.0
+
+:p 
+  x : Float = runTest
+  x
+> Type error:Multiple candidate class dictionaries for: (InterfaceSynthesisTest Float32)
+>   - (instance77 : (InterfaceSynthesisTest Float32))
+>   - (instance78 : (InterfaceSynthesisTest Float32))
+>
+>   x : Float = runTest
+>               ^^^^^^^
+
+instance [InterfaceSynthesisTest Float] InterfaceSynthesisTest Int
+  runTest = 2
+
+-- error shows Float32, since that is where the duplicate instance arises
+:p 
+  x : Int = runTest
+  x
+> Type error:Multiple candidate class dictionaries for: (InterfaceSynthesisTest Float32)
+>   - (instance77 : (InterfaceSynthesisTest Float32))
+>   - (instance78 : (InterfaceSynthesisTest Float32))
```

Comment:
IMO displaying auto-generated names is even worse than not displaying any names. We should really refine it to talk about source locations

Reply:
Yeah, this is uglier than I was expecting. I can play around with it to see if we can add source locations (but I'm not sure we expose enough information about filenames to type inference to make that work?)

Reply:
Seems like it's hard to pass source info through, so I'll just remove the names for now.